processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -168,13 +180,13 @@ func New(
 		return nil, err
 	}
 
-	c.namespaceInformer = newNamespaceInformer(c.kc)
+	c.namespaceInformer = newNamespaceInformer(c.kc, c.stopCh)
```

Comment:
What bugs me with the current state of this change is that the `new*Informer()` methods not only create a new informer but also start it. These are also called from the `New()` method that creates the k8s Client. 

Is the plan to further refactor this in the upcoming follow-ups? The constructors should not be responsible for starting the Informers.

My understanding of the end goal is that the kube Client and its informers will be initialised once, and then get started once when the first processor's instance gets started (the rest will re-use the already started one), right?

If we agree on this, let's add some todo comments here so as to make it clear for the future PRs.

Reply:
The idea is that the client gets an informer that is already running, and by closing the stop channel, it signals that it doesn't want to use it anymore. There is no idempotent `Start` method for informers, I'd have to add our own wrapper around them, which I'd rather avoid.

I *could* move obtaining the informers completely into `Start`. There aren't really a lot of benefits to doing it in `New`, as the construction methods can't return errors anyway. Does that sound better?

Reply:
Can we clarify at which point the informers get started, and how they end up being attached to a specific client of a specific `k8sattributes` processor instance?

If I don't miss any details here it sounds like the shared informers should be created and already be in running state when the `Start` will be called. Or, the first `Start` that will be called takes the responsibility to start them and then the following ones re-use the existing informers (when applicable). Is this correct? 

Can we decide on how we plan to apply this logic, to avoid back and forth in follow up PRs?

Reply:
Sure, let's move this discussion to the [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36234). I'll describe what I have in mind there.

Reply:
I've described how I intend this to work in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36234#issuecomment-2866377624.